src/CommandQoSController.cpp

```diff
@@ -0,0 +1,133 @@
+#include "CommandQoSController.hpp"
+#include "CommandRunner.hpp"
+#include "Helpers.hpp"
+
+#include <google/protobuf/util/json_util.h>
+
+#include <process/defer.hpp>
+#include <process/dispatch.hpp>
+
+using ::mesos::modules::Module;
+using ::mesos::ResourceUsage;
+using ::mesos::slave::QoSCorrection;
+
+using std::list;
+using std::string;
+
+using process::Future;
+using process::Failure;
+
+namespace criteo {
+
+namespace mesos {
+
+class CommandQoSControllerProcess
+    : public process::Process<CommandQoSControllerProcess> {
+ public:
+  CommandQoSControllerProcess(
+      const lambda::function<Future<ResourceUsage>()>& usage,
+      const Option<Command>& correctionsCommand, bool isDebugMode);
+
+  virtual process::Future<list<::mesos::slave::QoSCorrection>> corrections();
+  virtual list<::mesos::slave::QoSCorrection> m_corrections(
+      const ResourceUsage& m_usage);
+  virtual inline const Option<Command>& correctionsCommand() const {
+    return m_correctionsCommand;
+  }
+
+ protected:
+  const lambda::function<Future<ResourceUsage>()> m_usage;
+
+ private:
+  const Option<Command> m_correctionsCommand;
+  bool m_isDebugMode;
+};
+
+CommandQoSControllerProcess::CommandQoSControllerProcess(
+    const lambda::function<process::Future<ResourceUsage>()>& usage,
+    const Option<Command>& correctionsCommand, bool isDebugMode)
+    : m_usage(usage),
+      m_correctionsCommand(correctionsCommand),
+      m_isDebugMode(isDebugMode) {}
+
+list<::mesos::slave::QoSCorrection> CommandQoSControllerProcess::m_corrections(
+    const ResourceUsage& m_usage) {
+  LOG(INFO) << "Start Corrections timer_start_module QoSController";
```

Comment:
this would be more appropriate for debug level I think

Reply:
Yes this is more for debug, It is use to show time taken between different part of the module. like: start of module, start of the script, end of script, end of module